dace/codegen/targets/fpga.py

```diff
@@ -1021,9 +1022,11 @@ def _generate_MapEntry(self, sdfg, dfg, state_id, node, function_stream,
             # Generate nested loops
             if not isinstance(node, dace.sdfg.nodes.PipelineEntry):
 
-                if is_innermost and not fully_degenerate:
+                if is_innermost and not fully_degenerate and not is_degenerate[0]:
```

Comment:
I don't understand this condition: why not put it if only the outer loop is degenerate?

Actually it would make more sense if the condition was "if more than one loop is *not* degenerate".

Reply:
Because that was generating something like the following:
```
#pragma loop_coalesce
int index=0; //degenerate loop
for(....)
```
and intel compiler complains because there is a loop_coalesce but there is no loop. If some loop in the middle of the loop nest is generated then it would be ok.
```
# pragma loop_coalesce
for(....)
  for(....)
    int idx =0 ; //degenerate loop
```

Reply:
Ok, but then we need to move the coalesce pragma to the first non-degenerate loop instead of removing it completely, no?